test/changelog_web/controllers/feed_controller_test.exs

```diff
@@ -140,4 +140,93 @@ defmodule ChangelogWeb.FeedControllerTest do
+    setup [:setup_topic_feed_tests]
+
+    test "the topic feed", %{
+      conn: conn,
+      topics: topics,
+      news_items_with_topics: news_items_with_topics,
+      posts_with_topics: posts_with_topics,
+      podcast_episodes_with_topics: podcast_episodes_with_topics
+    } do
+      topic_slug = "rust"
+      conn = get(conn, Routes.feed_path(conn, :topic, topic_slug))
+
+      assert conn.status == 200
+      assert valid_xml(conn)
+    end
+
+    test "the topic's news feed", %{
+      conn: conn,
+      topics: topics,
+      news_items_with_topics: news_items_with_topics,
+      posts_with_topics: posts_with_topics,
+      podcast_episodes_with_topics: podcast_episodes_with_topics
+    } do
+      topic_slug = "rust"
+      conn = get(conn, Routes.feed_path(conn, :topic_news, topic_slug))
+
+      assert conn.status == 200
+      assert valid_xml(conn)
+    end
+
+    test "the topic's podcasts feed", %{
+      conn: conn,
+      topics: topics,
+      news_items_with_topics: news_items_with_topics,
+      posts_with_topics: posts_with_topics,
+      podcast_episodes_with_topics: podcast_episodes_with_topics
+    } do
+      topic_slug = "rust"
+      conn = get(conn, Routes.feed_path(conn, :topic_podcasts, topic_slug))
+
+      assert conn.status == 200
+      assert valid_xml(conn)
+
+      topic = topics[topic_slug]
+      assert conn.resp_body =~ topic.name
+      assert conn.resp_body =~ topic.description
+
+      Enum.each(posts_with_topics, fn post_with_topic ->
+        assert conn.resp_body =~ post_with_topic.post.title
+        assert conn.resp_body =~ post_with_topic.post.body
+      end)
+
+      Enum.each(podcast_episodes_with_topics, fn podcast_episode_with_topic ->
+        assert conn.resp_body =~ podcast_episode_with_topic.episode.title
+        assert conn.resp_body =~ podcast_episode_with_topic.episode.summary
+      end)
+    end
+  end
+
+  defp setup_topic_feed_tests(context) do
+    # Add some topics
+    t1 = insert(:topic, slug: "rust", name: "Rust", description: "Rust is a systems programming language created by Mozilla.")
+    # Create some news items
+    ni1 = insert(:published_news_item, headline: "Rust is pretty cool", story: "Use it or lose it")
+    # Associate those news items with topics
+    nit1 = insert(:news_item_topic, topic: t1, news_item: ni1)
+
+    # Add a post
+    post1 = insert(:published_post, body: "Can you believe this Rust thing?!")
+    pt1 = insert(:post_topic, post: post1, topic: t1)
+
+    # Add a podcast
+    pod1 = insert(:podcast, name: "Rust Crust", slug: "rustcrust", description: "Rust is neat")
+    # Add a podcast episode
+    e1 = insert(:published_episode, podcast: pod1)
+    # Associate that episode with a topic
+    et1 = insert(:episode_topic, episode: e1, topic: t1)
+
+    %{
+      topics: %{
+        "rust" => t1
+      },
+      news_items_with_topics: [nit1],
+      posts_with_topics: [pt1],
+      podcast_episodes_with_topics: [et1]
+    }
+  end
```

Comment:
```
  1) test Topic feed the topic's podcasts feed (ChangelogWeb.FeedControllerTest)
     test/changelog_web/controllers/feed_controller_test.exs:175
     Assertion with =~ failed
     code:  assert conn.resp_body =~ post_with_topic.post.title
     left:  "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<rss version=\"2.0\" xmlns:dc=\"http://purl.org/dc/elements/1.1/\" xmlns:atom=\"http://www.w3.org/2005/Atom\" xmlns:itunes=\"http://www.itunes.com/dtds/podcast-1.0.dtd\">\n  <channel>\n    <title>Changelog: Rust</title>\n    <copyright>All rights reserved</copyright>\n    <link>http://localhost:4001/</link>\n    <atom:link href=\"http://localhost:4001/feed\" rel=\"self\" type=\"application/rss+xml\" />\n    <atom:link href=\"http://localhost:4001/\" rel=\"alternate\" type=\"text/html\" />\n    <language>en-us</language>\n    <description>Rust is a systems programming language created by Mozilla.</description>\n  \n  </channel>\n</rss>\n"
     right: "Post 4"
     stacktrace:
       test/changelog_web/controllers/feed_controller_test.exs:193: anonymous fn/2 in ChangelogWeb.FeedControllerTest."test Topic feed the topic's podcasts feed"/1
       (elixir 1.13.4) lib/enum.ex:937: Enum."-each/2-lists^foreach/1-0-"/2
       test/changelog_web/controllers/feed_controller_test.exs:192: (test)
```
Currently trying to figure out why this test is failing. I think my test setup with the various `ExMachina` `insert` statements isn't set up correctly. 🤔 I'll hack at it some more soon!